Cabal/Distribution/Simple/Test/LibV09.hs

```diff
@@ -212,7 +213,12 @@ stubMain :: IO [Test] -> IO ()
 stubMain tests = do
     (f, n) <- fmap read getContents
     dir <- getCurrentDirectory
-    results <- tests >>= stubRunTests
+    let errhandler :: SomeException -> IO TestLogs
+        errhandler e = do
+          return $ TestLog { testName = "Cabal test suite exception",
+                             testOptionsReturned = [],
+                             testResult = Error $ displayException e }
```

Comment:
Oh, also, are we really sure we should catch all exceptions here, including async exceptions? What do we expect to happen if the user hits ctl-C during the test run? That should not be reported as a test suite exception right, that should just stop.

If I'm right about that then this handler ought to work like this: catch all exception types *except for* async exceptions. We can do that like so:

    case fromException e of
      Just UserInterrupt -> throwIO e -- let ctl-C propagate
      _ -> return TestLog {.... etc